test/e2e_env/universal/trafficroute/traffic_route.go

```diff
@@ -677,40 +687,41 @@ destinations:
       kuma.io/service: test-server
 conf:
   http:
+  - match: # this match is here to be able to check if the config was set
+      path:
+        prefix: /i-am-here
+    modify:
+      requestHeaders:
+        add:
+        - name: X-I-Am-Here
+          value: 'route-internal-external'
+    destination:
+      kuma.io/service: test-server
   - match:
       path:
         prefix: /
     split:
-    - weight: 50
+    - weight: 1
       destination:
         kuma.io/service: test-server
         version: v1
-    - weight: 50
+    - weight: 1
       destination:
         kuma.io/service: route-es-http
   destination:
     kuma.io/service: test-server
 `)(universal.Cluster)).To(Succeed())
-			Expect(universal.Cluster.Install(YamlUniversal(fmt.Sprintf(`
-type: ExternalService
-name: route-es-http-1
-mesh: trafficroute
-networking:
-  address: %s
-tags:
-  kuma.io/service: route-es-http
-  kuma.io/protocol: http
-`, esHttpHostPort)))).To(Succeed())
 
-			Eventually(func() (map[string]int, error) {
-				return client.CollectResponsesByInstance(universal.Cluster, "demo-client", "test-server.mesh")
-			}, "30s", "500ms").Should(
-				And(
-					HaveLen(2),
-					HaveKey(Equal(`echo-v1`)),
-					HaveKey(Equal(`route-es-http`)),
-				),
-			)
+			// Check and retry until the config got propagated to the client
+			Eventually(func() ([]server_types.EchoResponse, error) {
+				return client.CollectResponses(universal.Cluster, "demo-client", "test-server.mesh/i-am-here")
+			}, "1m", "500ms").MustPassRepeatedly(5).Should(HaveEach(HaveField("Received.Headers", HaveKeyWithValue("X-I-Am-Here", []string{"route-internal-external"}))))
+
+			Expect(client.CollectResponsesByInstance(universal.Cluster, "demo-client", "test-server.mesh", client.WithNumberOfRequests(100))).
+				Should(And(
+					HaveKeyWithValue("echo-v1", BeNumerically("~", 50, 10)),
```

Comment:
I dunno, is `10` enough? Why not `20`, `30`? Like why should there be ever an error where the weights are distorted by say 50%? The only error I cna imagine would be one where one of the weights is set to `0`

Reply:
We could do just HaveKey but why not do the check too? Envoy is still supposed to respect some level of Weighted routing